sds/src/scanner/mod.rs

```diff
@@ -2525,6 +2527,203 @@ mod test {
             _ => assert!(false),
         }
     }
+    #[cfg(feature = "match_validation")]
+    #[tokio::test]
```

Comment:
I'm glad you found the `#[tokio::test]` which is very helpful here. Just keep in mind that by default this creates a single-threaded tokio runtime.  It may not be obvious what's going on if you didn't know that. The singled-threaded one is fine most of the time, but if you ever find a test that seems to deadlock / stop progress and you can't figure out why, it's usually because of that.
To switch, use `#[tokio::test(flavor = "multi_thread")]`

Reply:
good to know thanks